src/project.ts

```diff
@@ -0,0 +1,214 @@
+import { MapObjectLayers } from "./map";
+import { loadBitmapAsync, requestJSONAsync } from "./util";
+
+export class Project {
+    images: ProjectImage[];
+    templates: MapObjectTemplate[][];
+    tiles: ProjectSprite[];
+    tileSize = 16;
+    canvas: HTMLCanvasElement;
+
+    protected tileColors: string[];
+
+    constructor() {
+        this.images = [];
+        this.templates = [[], [], [], [], [], [], []];
```

Comment:
this has seven arrays, but there are currently only 5 MapObjectLayers; are the extras for other planned layers?